### DISCUSSION TITLE
Azure Deployment Slots for (at least) the Live Environment

### DISCUSSION THREAD

**Discussion by LennardF1989** [umbraco/Umbraco.Cloud.Issues#1012] [Ideas]
We're currently launching a new website on Umbraco Cloud, but I noticed there is no way to pre-stage the live environment in the sense I can keep my current website running, prepare the "new" live environment (push from stage to live, do final checkups, make sure things are really all A-OK) then swap the current live out with my new live environment.

Are there any plans to make this available? It would really make releasing a lot more hassle free. Users of the website can now suddenly be presented with an Umbraco installation/upgrade screen, and/or mess with the website as you are trying to get into a presentable state.

Just to clarify upfront: Using the current Stage environment does not solve this particular issue, only that of the content preparation (to a certain extend)! Releasing a website, which requires Apppool recycles, Umbraco Deploy imports, potential DB upgrades and what not cannot be prepared on a Stage environment and flawlessly pushed to Live as-is.

**Comment by sitereactor** [comment]
Hey, sorry that this feature request has been left without an answer for so long.
Unfortunately this is not something we are able to do in our current Umbraco Cloud setup, but good news is that we are in the process of modernizing our infrastructure to enable capabilities such as deployment slots. It's something that has been on my own wishlist for years, so keen to see it added - hopefully this calendar year.

We'll keep this as a feature request and update once its planned as part of the Now/Next/Later stages of the Roadmap.

**Comment by paulsterling** [comment]
According to @shazwazza there will likely be an issue with the Examine indexes when using slots to swap sites.

https://github.com/Shazwazza/Examine/issues/382

We currently see the `Lucene.Net.Index.CorruptIndexException: invalid deletion count` issue occassionally on our Umbraco Cloud sites (13.4.0) even without using deployment slots.

I know that @shazwazza has created https://examinex.online/ to address the issue with Examine/Lucene and Azure. Possibly there is a way to license that product and bundle with Cloud...or to offer it as an option?

/cc @ksuvec

**Comment by Amalie-Wowern** [comment]
This is really a feature we are missing in Umbraco cloud

And maybe the problem with examine is fixed with the latest release?

**Comment by paulsterling** [comment]
Just a note that the most recent Examine release (3.3.x) likely addresses the above issue.

https://github.com/Shazwazza/Examine/releases/tag/v3.3.0
